pkg/controller/operators/olm/operator.go

```diff
@@ -1241,14 +1249,112 @@ func (a *Operator) syncCopyCSV(obj interface{}) (syncError error) {
 	}).Debug("copying csv to targets")
 
 	// Check if we need to do any copying / annotation for the operatorgroup
-	if err := a.ensureCSVsInNamespaces(clusterServiceVersion, operatorGroup, NewNamespaceSet(operatorGroup.Status.Namespaces)); err != nil {
-		logger.WithError(err).Info("couldn't copy CSV to target namespaces")
-		syncError = err
+	namespaceSet := NewNamespaceSet(operatorGroup.Status.Namespaces)
+	if a.copiedCSVs || !namespaceSet.IsAllNamespaces() {
+		if err := a.ensureCSVsInNamespaces(clusterServiceVersion, operatorGroup, namespaceSet); err != nil {
+			logger.WithError(err).Info("couldn't copy CSV to target namespaces")
+			syncError = err
+		}
+
+		// If the CSV was installed in AllNamespace mode, remove any "CSV Copying Disabled" events
+		// in which the related object's name, namespace, and uid match the given CSV's.
+		if namespaceSet.IsAllNamespaces() {
+			if err := a.deleteCSVCopyingDisabledEvent(clusterServiceVersion); err != nil {
+				return err
+			}
+		}
+	} else {
+		requirement, err := labels.NewRequirement(operatorsv1alpha1.CopiedLabelKey, selection.Equals, []string{clusterServiceVersion.Namespace})
+		if err != nil {
+			return err
+		}
+
+		copiedCSVs, err := a.copiedCSVLister.List(labels.NewSelector().Add(*requirement))
+		if err != nil {
+			return err
+		}
+
+		for _, copiedCSV := range copiedCSVs {
+			err := a.client.OperatorsV1alpha1().ClusterServiceVersions(copiedCSV.Namespace).Delete(context.TODO(), copiedCSV.Name, metav1.DeleteOptions{})
+			if err != nil && !k8serrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+		if err := a.createCSVCopyingDisabledEvent(clusterServiceVersion); err != nil {
+			return err
+		}
 	}
 
 	return
 }
 
+func (a *Operator) getCopiedCSVDisabledEventsForCSV(csv *operatorsv1alpha1.ClusterServiceVersion) ([]corev1.Event, error) {
+	result := []corev1.Event{}
+	if csv == nil {
+		return result, nil
+	}
+
+	events, err := a.opClient.KubernetesInterface().CoreV1().Events(csv.GetNamespace()).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, event := range events.Items {
+		a.logger.Infof("For event %v\n", event)
```

Comment:
nit: Is there extra information for this log message we could include here? It may be difficult understanding what it conveys in a production system.

Reply:
oh haha this is from some debugging I had done, I'll be removing it.